### PR TITLE
Comments new search method

### PR DESCRIPTION
**Pull request by adudekyeti**
Comments new search method

### CODE REVIEW

File: modules/Vtiger/views/Detail.php [hunk 1]
```diff
@@ -587,16 +608,99 @@ public function showAllComments(\App\Request $request)
...
+			$viewer->assign('IS_READ_ONLY', $request->getBoolean('isReadOnly'));
+			$viewer->assign('NO_COMMENT_FORM', true);
+			$viewer->assign('MODULE', $moduleName);
+			if ($isWidget === false) {
+				$viewer->assign('SHOW_CHILD_COMMENTS', true);
+				return $viewer->view('CommentsList.tpl', $moduleName, true);
+			} else {
+				$viewer->assign('PAGING_MODEL', $pagingModel);
+				$viewer->assign('BUTTON_SHOW_PARENT', true);
+				return $viewer->view('CommentsList.tpl', $moduleName, true);
```
Comment (rskrzypczak) [inline modules/Vtiger/views/Detail.php]: duplicate: return $viewer->view('CommentsList.tpl', $moduleName, true);

File: modules/Vtiger/views/Detail.php [hunk 2]
```diff
@@ -587,16 +608,99 @@ public function showAllComments(\App\Request $request)
...
+			return $this->showAllComments($request);
+		} else {
+			$searchValue = $request->getByType('search_key', 'Text');
+			$parentCommentModels = ModComments_Record_Model::getSearchComments($recordId, $searchValue, $isWidget, $hierarchy, $pagingModel);
+		}
+		$viewer = $this->getViewer($request);
+		if (!empty($parentCommentModels)) {
+			$modCommentsModel = Vtiger_Module_Model::getInstance('ModComments');
+			$viewer->assign('COMMENTS_MODULE_MODEL', $modCommentsModel);
+			$viewer->assign('CURRENTUSER', $currentUserModel);
```
Comment (rskrzypczak) [inline modules/Vtiger/views/Detail.php]: remove

File: modules/Vtiger/views/Detail.php [hunk 3]
```diff
@@ -587,16 +608,99 @@ public function showAllComments(\App\Request $request)
...
+			$viewer->assign('PARENT_COMMENTS', $parentCommentModels);
+			$viewer->assign('CURRENT_COMMENT', false);
+			$viewer->assign('IS_READ_ONLY', $request->getBoolean('isReadOnly'));
+			$viewer->assign('NO_COMMENT_FORM', true);
+			$viewer->assign('MODULE', $moduleName);
+			if ($isWidget === false) {
+				$viewer->assign('SHOW_CHILD_COMMENTS', true);
+				return $viewer->view('CommentsList.tpl', $moduleName, true);
+			} else {
+				$viewer->assign('PAGING_MODEL', $pagingModel);
```
Comment (rskrzypczak) [inline modules/Vtiger/views/Detail.php]: ??

File: modules/Vtiger/views/Detail.php [hunk 4]
```diff
@@ -506,14 +500,43 @@ public function showChildComments(\App\Request $request)
...
+		}
+		$parentCommentModel = Vtiger_Record_Model::getInstanceById($request->getInteger('commentid'));
+		$parentThreadComments = $parentCommentModel->getParentComments();
+		$currentUserModel = Users_Record_Model::getCurrentUserModel();
+		$modCommentsModel = Vtiger_Module_Model::getInstance('ModComments');
+		$viewer = $this->getViewer($request);
+		$viewer->assign('PARENT_COMMENTS', $parentThreadComments);
+		$viewer->assign('SHOW_CHILD_COMMENTS', true);
+		$viewer->assign('NO_COMMENT_FORM', true);
+		$viewer->assign('CURRENTUSER', $currentUserModel);
```
Comment (rskrzypczak) [inline modules/Vtiger/views/Detail.php]: Remove

File: layouts/basic/modules/Vtiger/ShowAllComments.tpl [hunk 5]
```diff
@@ -32,80 +38,48 @@
...
-					<span class="input-group-prepend" id="commentSearchAddon">
-						<div class="input-group-text"><span class="fas fa-search"></span> </div>
-					</span>
-					<input type="text" class="form-control commentSearch" placeholder="{\App\Language::translate('LBL_COMMENTS_SEARCH','ModComments')}" aria-describedby="commentSearchAddon">
+		<div class="row">
+			<div class="col-lg-6"></div>
+			<div class="col-md-12 col-lg-6 form-row commentsHeader my-3">
+				<div class="col-6 col-lg-6 col-md-6 col-sm-6 p-0">
+					<div class="input-group-append bg-white rounded-right">
+						<input type="text" class="js-commentSearch form-control commentSearch"
```
Comment (rskrzypczak) [inline layouts/basic/modules/Vtiger/ShowAllComments.tpl]: js-commentSearch commentSearch
remove commentSearch or change to BEM

File: layouts/basic/modules/Vtiger/ShowAllComments.tpl [hunk 6]
```diff
@@ -32,80 +38,48 @@
...
+						<label class="btn btn-sm btn-outline-primary {if $HIERARCHY_VALUE !== 'all'}active{/if}">
+							<input class="js-detailHierarchyComments detailHierarchyComments" type="radio"
+								   name="options" id="option1"
+								   value="current" autocomplete="off"
+								   {if $HIERARCHY_VALUE !== 'all'}checked="checked"{/if}
+								   data-js="change"
+							> {\App\Language::translate('LBL_COMMENTS_0', 'ModComments')}
+						</label>
+						<label class="btn btn-sm btn-outline-primary {if $HIERARCHY_VALUE === 'all'}active{/if}">
+							<input class="js-detailHierarchyComments detailHierarchyComments" type="radio"
```
Comment (rskrzypczak) [inline layouts/basic/modules/Vtiger/ShowAllComments.tpl]: remove detailHierarchyComments or change to BEM
**and many, many similar elements**

File: modules/Vtiger/views/Detail.php [hunk 7]
```diff
@@ -587,16 +607,87 @@ public function showAllComments(\App\Request $request)
...
+			$viewer->assign('CURRENTUSER', $currentUserModel);
+			$viewer->assign('PARENT_COMMENTS', $parentCommentModels);
+			$viewer->assign('CURRENT_COMMENT', false);
+			$viewer->assign('IS_READ_ONLY', $request->getBoolean('isReadOnly'));
+			$viewer->assign('NO_COMMENT_FORM', true);
+			$viewer->assign('MODULE', $moduleName);
+			if ($isWidget === false) {
+				$viewer->assign('SHOW_CHILD_COMMENTS', true);
+			} else {
+				$viewer->assign('BUTTON_SHOW_PARENT', true);
```
Comment (rskrzypczak) [inline modules/Vtiger/views/Detail.php]: maybe just one variable (SHOW_CHILD_COMMENTS or BUTTON_SHOW_PARENT)

File: modules/Vtiger/views/Detail.php (hunk 4 above)
Comment (mariuszkrzaczkowski) [reply modules/Vtiger/views/Detail.php]: ??? CURRENTUSER

File: modules/Vtiger/views/Detail.php (hunk 2 above)
Comment (mariuszkrzaczkowski) [reply modules/Vtiger/views/Detail.php]: ??  CURRENTUSER

File: modules/Vtiger/views/Detail.php (hunk 7 above)
Comment (mariuszkrzaczkowski) [reply modules/Vtiger/views/Detail.php]: ??

File: public_html/layouts/basic/modules/Vtiger/resources/Detail.js [hunk 8]
```diff
@@ -1784,6 +1784,19 @@ jQuery.Class("Vtiger_Detail_Js", {
 		});
 		return aDeferred.promise();
 	},
+	/**
+	 * Function to get parent comment
+	 * @param {number} commentId
+	 * @returns {string}
+	 */
+	getParentComments(commentId) {
+		let aDeferred = jQuery.Deferred(),
```
Comment (mariuszkrzaczkowski) [inline public_html/layouts/basic/modules/Vtiger/resources/Detail.js]: jQuery>> $

File: public_html/layouts/basic/modules/Vtiger/resources/Detail.js [hunk 9]
```diff
@@ -1927,105 +1940,102 @@ jQuery.Class("Vtiger_Detail_Js", {
...
-			var addCommentBlock = thisInstance.getCommentBlock();
-			addCommentBlock.appendTo('.commentBlock');
+			self.removeCommentBlockIfExists();
+			self.getCommentBlock().appendTo('.commentBlock');
 		});
 		detailContentsHolder.on('click', '.closeCommentBlock', function (e) {
-			var currentTarget = jQuery(e.currentTarget);
-			var commentInfoBlock = currentTarget.closest('.singleComment');
-			commentInfoBlock.find('.commentActionsContainer').show();
+			let commentInfoBlock = jQuery(e.currentTarget.closest('.js-comment-single'));
```
Comment (mariuszkrzaczkowski) [inline public_html/layouts/basic/modules/Vtiger/resources/Detail.js]: jQuery>> $

File: public_html/layouts/basic/modules/Vtiger/resources/Detail.js [hunk 10]
```diff
@@ -1927,105 +1940,102 @@ jQuery.Class("Vtiger_Detail_Js", {
...
+			commentInfoBlock.find('.js-comment-container').show();
 			commentInfoBlock.find('.commentInfoContent').show();
```
Comment (mariuszkrzaczkowski) [inline public_html/layouts/basic/modules/Vtiger/resources/Detail.js]: .commentInfoContent ???

File: public_html/layouts/basic/modules/Vtiger/resources/Detail.js [hunk 11]
```diff
@@ -1927,105 +1940,102 @@ jQuery.Class("Vtiger_Detail_Js", {
...
+	 * @param {jQuery} detailContentsHolder
+	 */
+	registerCommentEvents(detailContentsHolder) {
+		const self = this;
 		detailContentsHolder.on('click', '.addCommentBtn', function (e) {
-			thisInstance.removeCommentBlockIfExists();
-			var addCommentBlock = thisInstance.getCommentBlock();
-			addCommentBlock.appendTo('.commentBlock');
+			self.removeCommentBlockIfExists();
+			self.getCommentBlock().appendTo('.commentBlock');
```
Comment (mariuszkrzaczkowski) [inline public_html/layouts/basic/modules/Vtiger/resources/Detail.js]: .commentBlock  ???

**Pull request merged**